app/forms/gobierto_admin/gobierto_common/vocabulary_terms_import_form.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module GobiertoAdmin
+  module GobiertoCommon
+    class VocabularyTermsImportForm < BaseForm
+      class CSVRowInvalid < ArgumentError; end
+      REQUIRED_COLUMNS = %w(term_id parent_id term_name_ES term_name_CA term_description_ES term_description_CA).freeze
+
+      attr_accessor(
+        :vocabulary,
+        :csv_file
+      )
+
+      validates :vocabulary, :csv_file, presence: true
+      validate :csv_format
+
+      delegate :terms, to: :vocabulary
+
+      def save
+        import if valid?
+      end
+
+      private
+
+      def file_raw_content
+        return @csv_file.open unless @csv_file.is_a? String
+
+        File.open(@csv_file)
+      end
+
+      def import
+        if csv_file.present?
+          ActiveRecord::Base.transaction do
+            clear_previous_terms
+            import_terms_from_csv
+          end
+        end
+      rescue CSVRowInvalid => e
+        errors.add(:base, :invalid_row, row_data: e.message)
+        false
+      end
+
+      def clear_previous_terms
+        terms.destroy_all
+      end
+
+      def import_terms_from_csv
+        position = 0
+
+        csv_content.each do |row|
+          position += 1
+          term_form = TermForm.new(
```

Comment:
Not a big deal but many of those strips can be avoided if you use [CSV Converters](https://ruby-doc.org/stdlib-2.6.3/libdoc/csv/rdoc/CSV.html) and [Ruby guide - CSV](https://www.rubyguides.com/2018/10/parse-csv-ruby/). They are nice for this kind of situation

Reply:
Very good point. I'll consider this in the future. Thanks!